src/ParallelKernel/allocators.jl

```diff
@@ -22,7 +34,13 @@ macro ones(args...) check_initialized(); esc(_ones(args...)); end
 const RAND_DOC = """
     @rand(args...)
 
+!!! note "Advanced"
+        @rand(numbertype, args...)
+
 Call `rand(numbertype, args...)`, where `numbertype` is the datatype selected with [`@init_parallel_kernel`](@ref) and the function `rand` is chosen/implemented to be compatible with the package for parallelization selected with [`@init_parallel_kernel`](@ref).
+
+!!! note "Advanced"
+    The `numbertype` can be explicitly passed as argument in order to be used instead of the default `numbertype` chosen with [`@init_parallel_kernel`](@ref). If no default `numbertype` was chosen [`@init_parallel_kernel`](@ref), then the argument `numbertype` is mandatory. This needs to be used with care to ensure that no datatype conversions occurs in performance critical computations.
 """
```

Comment:
Would it make sense to add a `fill` macro with doc to be displayed here?

Reply:
in the next PR